awscli/clidriver.py

```diff
@@ -106,18 +108,32 @@ def _get_distribution_source():
     return 'bundle'
 
 
-def _get_human_readable_os_name():
-    return 'os_name'
+def _get_human_readable_linux_distribution():
+    linux = None
+    try:
+        distribution = distro.id()
+        version = distro.major_version()
+        if distribution and distribution != 'darwin':
+            linux = distribution
+            if version:
+                linux += '.%s' % version
+    except Exception:
+        linux = 'linux'
+    return linux
```

Comment:
So for this logic, I was thinking something more like this:
```python
def _get_distribution(self):
    distribution = None
    if platform.system() == 'Linux':
        distribution = self._get_linux_distribution()
    return distribution

def _get_linux_distribution(self):
    linux_distribution = None
    try:
        linux_distribution = distro.id()
        version = distro.major_version()
         if version:
             linux_distribution += '.%s' % version
    except Exception:
        return linux_distribution
```
The main changes that I was thinking:
1. It is a bit more defensive for non-Linux systems in that we do not even call the `distro` library for Windows and Mac. While there was an if statement for mac in the PR, I'm not sure if we can guarantee that the if statement will still apply if we upgrade versions of the library and say they add more support for non-Linux systems.
2. If for whatever reason we cannot get a major version for the linux distribtution, we would still be able to include the distribution id.
3. If we cannot figure out the distribution in general, we should just not add it to the user agent. The general `linux` value is a bit of an ambiguous value if you are not familiar with the code to know that means the distribution is unknown. I would be fine with either just not including it or have a more specific value like `unknown`.